subsys/net/lib/sockets/sockets.c

```diff
@@ -168,3 +178,67 @@ ssize_t zsock_send(int sock, const void *buf, size_t len, int flags)
 
 	return len;
 }
+
+ssize_t zsock_recv(int sock, void *buf, size_t max_len, int flags)
+{
+	(void)flags;
+	struct net_context *ctx = INT_TO_POINTER(sock);
+	enum net_sock_type sock_type = net_context_get_type(ctx);
+	size_t recv_len = 0;
+
+	if (sock_type == SOCK_DGRAM) {
+		__ASSERT(0, "DGRAM is not yet handled");
+	} else if (sock_type == SOCK_STREAM) {
+		do {
```

Comment:
I can. So, should I? (I don't expect source-level issues, like need to re-do some work in this function, but I'm not sure if a compiler will readily inline such a function, which in turn may lead to higher stack usage, etc.)